### PR TITLE
Some delete work

### PR DESCRIPTION
**Pull request by SeleniumK**

### CODE REVIEW

File: test_bst.py [hunk 1]
```diff
@@ -266,3 +284,132 @@ def test_empty_tree_traversals(empty_bst):
...
+
+
+def test_delete_middle(bst_root_fifty):
+    bst_root_fifty.delete(25)
+    # import pdb; pdb.set_trace()
+    assert bst_root_fifty.root.value == 50
+    assert bst_root_fifty.root.left_child.value == 35 
+    assert bst_root_fifty.root.left_child.left_child is None
+    assert bst_root_fifty.root.left_child.right_child.value == 35
+    assert bst_root_fifty.root.left_child.right_child.parent.value == 12
```
Comment (Jakeand3rson) [inline test_bst.py]: So the left child's value?

File: test_bst.py [hunk 2]
```diff
@@ -266,3 +284,132 @@ def test_empty_tree_traversals(empty_bst):
...
+        assert not bst_root_fifty.contains(num)
+    assert bst_root_fifty.depth() == 2
+    for num in [25, 100]:
+        assert bst_root_fifty._contains(num).left_child is None
+        assert bst_root_fifty._contains(num).right_child is None
+
+
+def test_delete_middle(bst_root_fifty):
+    bst_root_fifty.delete(25)
+    # import pdb; pdb.set_trace()
```
Comment (Jakeand3rson) [inline test_bst.py]: This shouldn't be here...

File: test_bst.py [hunk 3]
```diff
@@ -266,3 +284,132 @@ def test_empty_tree_traversals(empty_bst):
...
+
+def test_delete_root(bst_root_fifty):
+    bst_root_fifty.delete(50)
+    root = bst_root_fifty.root
+    assert root.value == 25
+    assert root.right_child.value == 100
+    assert root.left_child.value == 12
+    assert root.right_child.right_child.value == 35
+    assert root.right_child.right_child.value == 150
+    assert root.right_child.left_child.value == 75
```
Comment (Jakeand3rson) [inline test_bst.py]: There is a boatload of assert's here. Can you think of a way to break these down?

File: test_bst.py [hunk 4]
```diff
@@ -266,3 +284,132 @@ def test_empty_tree_traversals(empty_bst):
...
+#     assert root_left_child.left_child.value == 13
+#     assert root_left_child.left_child.left_child.value == 6
+
+
+# def test_two_node_bst(empty_bst):
+#     empty_bst.insert(100)
+#     empty_bst.insert(75)
+#     empty_bst.insert(150)
+#     empty_bst.delete(100)
+#     assert empty_bst.root.value == 150
```
Comment (Jakeand3rson) [inline test_bst.py]: If you aren't using these, you should not have these in here. At least not without documenting it really well so your readers know why they aren't being included.

File: test_bst.py (hunk 1 above)
Comment (qwergram) [inline test_bst.py]: Right, we just wanted to assert the parent's connection to the left child was being created properly.

**Pull request merged**